### PR TITLE
COMPAT: Support for MPL 1.5

### LINKED ISSUE #11111: COMPAT: matplotlib v0.15rc1 issues
install via: ``conda install matplotlib -c tacaswell`` (only linux-64 ATM)

this is with ``pandas-0.17rc1`` 

```
......F......E..FF...FE..F./home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/__init__.py:872: UserWarning: axes.color_cycle is deprecated and replaced with axes.prop_cycle; please use the latter.
  warnings.warn(self.msg_depr % (key, alt_key))
/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/__init__.py:892: UserWarning: axes.color_cycle is deprecated and replaced with axes.prop_cycle; please use the latter.
  warnings.warn(self.msg_depr % (key, alt_key))
...E.../home/jreback/pandas/pandas/tools/plotting.py:3242: UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared
  "is being cleared", UserWarning)
EF......ES.....E.......E.E.........../home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/artist.py:221: MatplotlibDeprecationWarning: This has been deprecated in mpl 1.5, please use the
axes property.  A removal date has not been set.
  warnings.warn(_get_axes_msg, mplDeprecation, stacklevel=1)
.........FE..E........F.E..E..........E.......................EE.
======================================================================
ERROR: test_bar_colors (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1810, in test_bar_colors
    self._check_colors(ax.patches[::5], facecolors=default_colors[:5])
TypeError: 'NoneType' object has no attribute '__getitem__'

======================================================================
ERROR: test_boxplot_colors (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2860, in test_boxplot_colors
    _check_colors(bp, default_colors[0], default_colors[0], default_colors[2])
TypeError: 'NoneType' object has no attribute '__getitem__'

======================================================================
ERROR: test_df_legend_labels (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2478, in test_df_legend_labels
    ax = df.plot(legend=True, secondary_y='b')
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3610, in __call__
    sort_columns=sort_columns, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2495, in plot_frame
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1596, in _make_plot
    ax = self._get_ax(i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_errorbar_plot (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 3129, in test_errorbar_plot
    df.plot(yerr=df_err)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3610, in __call__
    sort_columns=sort_columns, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2495, in plot_frame
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1609, in _make_plot
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1623, in _plot
    lines = MPLPlot._plot(ax, x, y_values, style=style, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1241, in _plot
    return ax.errorbar(x, y, **kwds)
  File "/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/__init__.py", line 1806, in inner
    return func(ax, *args, **kwargs)
  File "/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/axes/_axes.py", line 2964, in errorbar
    raise ValueError("yerr must be a scalar, the same "
ValueError: yerr must be a scalar, the same dimensions as y, or 2xN.

======================================================================
ERROR: test_hist_colors (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2737, in test_hist_colors
    self._check_colors(ax.patches[::10], facecolors=default_colors[:5])
TypeError: 'NoneType' object has no attribute '__getitem__'

======================================================================
ERROR: test_kde_colors_and_styles_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2799, in test_kde_colors_and_styles_subplots
    for ax, c in zip(axes, list(default_colors)):
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_line_colors_and_styles_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2634, in test_line_colors_and_styles_subplots
    for ax, c in zip(axes, list(default_colors)):
TypeError: 'NoneType' object is not iterable

======================================================================
ERROR: test_line_lim (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1768, in test_line_lim
    ax = df.plot(secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3610, in __call__
    sort_columns=sort_columns, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2495, in plot_frame
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_subplots_dup_columns (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1668, in test_subplots_dup_columns
    axes = df.plot(subplots=True, secondary_y='a')
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3610, in __call__
    sort_columns=sort_columns, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2495, in plot_frame
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1276, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_subplots_sharex_axes_existing_axes (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1651, in test_subplots_sharex_axes_existing_axes
    df['C'].plot(ax=axes[0], secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_df_series_secondary_legend (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 889, in test_df_series_secondary_legend
    s.plot(legend=True, secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_errorbar_plot (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1113, in test_errorbar_plot
    s.plot(yerr=s_err)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1609, in _make_plot
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1623, in _plot
    lines = MPLPlot._plot(ax, x, y_values, style=style, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1241, in _plot
    return ax.errorbar(x, y, **kwds)
  File "/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/__init__.py", line 1806, in inner
    return func(ax, *args, **kwargs)
  File "/home/jreback/miniconda/envs/pandas/lib/python2.7/site-packages/matplotlib/axes/_axes.py", line 2964, in errorbar
    raise ValueError("yerr must be a scalar, the same "
ValueError: yerr must be a scalar, the same dimensions as y, or 2xN.

======================================================================
ERROR: test_hist_secondary_legend (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 852, in test_hist_secondary_legend
    df['b'].plot.hist(ax=ax, legend=True, secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3524, in hist
    return self(kind='hist', bins=bins, **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1948, in _make_plot
    ax = self._get_ax(i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_ts_area_lim (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 582, in test_ts_area_lim
    ax = tz_ts.plot.area(stacked=False, secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3560, in area
    return self(kind='area', **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
ERROR: test_ts_line_lim (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 551, in test_ts_line_lim
    ax = self.ts.plot(secondary_y=True)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 3436, in __call__
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2526, in plot_series
    **kwds)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 2323, in _plot
    plot_obj.generate()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 944, in generate
    self._make_plot()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1579, in _make_plot
    if self._is_ts_plot():
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1572, in _is_ts_plot
    return not self.x_compat and self.use_index and self._use_dynamic_x()
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1576, in _use_dynamic_x
    return _use_dynamic_x(self._get_ax(0), self.data)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 1280, in _get_ax
    ax = self._maybe_right_yaxis(ax, i)
  File "/home/jreback/pandas/pandas/tools/plotting.py", line 976, in _maybe_right_yaxis
    new_ax._get_lines.color_cycle = orig_ax._get_lines.color_cycle
AttributeError: '_process_plot_var_args' object has no attribute 'color_cycle'

======================================================================
FAIL: test_area_colors (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2701, in test_area_colors
    self._check_colors(handles, linecolors=custom_colors)
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 194, in _check_colors
    self.assertEqual(len(collections), len(linecolors))
AssertionError: 10 != 5

======================================================================
FAIL: test_bar_log_no_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2188, in test_bar_log_no_subplots
    tm.assert_numpy_array_equal(ax.yaxis.get_ticklocs(), expected)
  File "/home/jreback/pandas/pandas/util/testing.py", line 822, in assert_numpy_array_equal
    left.shape, right.shape)
  File "/home/jreback/pandas/pandas/util/testing.py", line 791, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different

numpy array shapes are different
[left]:  (4,)
[right]: (2,)

======================================================================
FAIL: test_bar_log_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2199, in test_bar_log_subplots
    tm.assert_numpy_array_equal(ax[0].yaxis.get_ticklocs(), expected)
  File "/home/jreback/pandas/pandas/util/testing.py", line 822, in assert_numpy_array_equal
    left.shape, right.shape)
  File "/home/jreback/pandas/pandas/util/testing.py", line 791, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different

numpy array shapes are different
[left]:  (6,)
[right]: (4,)

======================================================================
FAIL: test_boxplot (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2214, in test_boxplot
    self.bp_n_objects * len(numeric_cols))
AssertionError: 21 != 24

======================================================================
FAIL: test_boxplot_vertical (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 2251, in test_boxplot_vertical
    self.assertEqual(len(ax.lines), self.bp_n_objects * len(numeric_cols))
AssertionError: 21 != 24

======================================================================
FAIL: test_errorbar_scatter (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 3251, in test_errorbar_scatter
    _check_errorbar_color(ax.containers, 'green', has_err='has_yerr')
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 3240, in _check_errorbar_color
    self._check_colors(errs, linecolors=[expected] * len(errs))
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 204, in _check_colors
    self.assertEqual(result, expected)
AssertionError: Tuples differ: (0.0, 0.0, 1.0, 1.0) != (0.0, 0.5019607843137255, 0.0,...

First differing element 1:
0.0
0.501960784314

- (0.0, 0.0, 1.0, 1.0)
+ (0.0, 0.5019607843137255, 0.0, 1.0)

======================================================================
FAIL: test_subplots (pandas.tests.test_graphics.TestDataFramePlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 1440, in test_subplots
    self._check_legend_labels(ax, labels=[com.pprint_thing(column)])
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 117, in _check_legend_labels
    self._check_text_labels(ax.get_legend().get_texts(), labels)
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 241, in _check_text_labels
    self.assertEqual(len(labels), len(expected))
AssertionError: 2 != 1

======================================================================
FAIL: test_bar_log (pandas.tests.test_graphics.TestSeriesPlots)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jreback/pandas/pandas/tests/test_graphics.py", line 649, in test_bar_log
    tm.assert_numpy_array_equal(ax.yaxis.get_ticklocs(), expected)
  File "/home/jreback/pandas/pandas/util/testing.py", line 822, in assert_numpy_array_equal
    left.shape, right.shape)
  File "/home/jreback/pandas/pandas/util/testing.py", line 791, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different

numpy array shapes are different
[left]:  (6,)
[right]: (4,)

----------------------------------------------------------------------
Ran 136 tests in 124.794s

FAILED (SKIP=1, errors=15, failures=8)
```

cc @TomAugspurger @sinhrks @jorisvandenbossche 
cc @tacaswell

### PR DESCRIPTION
**Pull request by TomAugspurger**
closes https://github.com/pydata/pandas/issues/11111  (nice issue number)

Testing with tox locally I get 1 failure using either matplotlib 1.4 or 1.5. I may have messed up my tox config though. Still need to cleanup a bunch of stuff, I just threw in changes to get each test passing as I went.

release notes et. al coming tonight or tomorrow. Are we making any changes to the build matrix to run a matplotlib 1.5?

### CODE REVIEW

**Comment by jreback** [comment]
u can add to the channel list (I think I showed this above)

it will thrn pick up the latest version if it's not specified exactly

File: pandas/tests/test_graphics.py [hunk 1]
```diff
@@ -74,13 +74,23 @@ def setUp(self):
...
                                       'category': random.randint(4, size=n)})
 
-        if str(mpl.__version__) >= LooseVersion('1.4'):
+        mpl_version = str(mpl.__version__)
+        if mpl_version >= LooseVersion('1.4') or mpl_version[0] == '0':
             self.bp_n_objects = 7
         else:
             self.bp_n_objects = 8
+        # Thier dev version is messed up, 0+unknown
+        if mpl_version[0] == '0' or mpl_version >= LooseVersion('1.5'):
```
Comment (jreback) [inline pandas/tests/test_graphics.py]: prob best to put these in some functions (either in ``util.testing`` or top-level here)

File: pandas/tests/test_graphics.py [hunk 2]
```diff
@@ -183,7 +193,7 @@ def _check_colors(self, collections, linecolors=None, facecolors=None,
         """
 
         from matplotlib.lines import Line2D
-        from matplotlib.collections import Collection
+        from matplotlib.collections import Collection, PolyCollection
```
Comment (jreback) [inline pandas/tests/test_graphics.py]: is this always availabel?

File: pandas/tools/plotting.py [hunk 3]
```diff
@@ -31,6 +31,16 @@
 except ImportError:
     pass
 
+try:
+    import matplotlib
+    mpl_v = matplotlib.__version__
+    if mpl_v >= LooseVersion('1.5') or mpl_v[0] == '0':
+        mpl_ge_1_5_0 = True
```
Comment (jreback) [inline pandas/tools/plotting.py]: definitly should be a function somewhere

File: pandas/tests/test_graphics.py (hunk 2 above)
Comment (TomAugspurger) [inline pandas/tests/test_graphics.py]: at leas as of 1.2.0, which I think is our oldest supported version.

Comment (jreback) [inline pandas/tests/test_graphics.py]: @TomAugspurger as an aside
if it makes things easier now (or in a later version )
we could move up the min supported version of matplotlib from 1.2
cc @tacaswell

Comment (tacaswell) [inline pandas/tests/test_graphics.py]: For reference, mpl does not support past the current minor release (ex, we currently do not fix bugs in 1.3.x, and in a short while will not fix bugs in 1.4.x).

File: pandas/core/config_init.py [hunk 4]
```diff
@@ -214,7 +214,8 @@
 
 def mpl_style_cb(key):
     import sys
-    from pandas.tools.plotting import mpl_stylesheet
+    from pandas.tools.plotting import mpl_stylesheet, _update_stylesheet
+    _update_stylesheet()
```
Comment (TomAugspurger) [inline pandas/core/config_init.py]: This is pretty hacky, but `mpl_stylesheet` is a dict defined in pandas.tools.plotting. I have my matplotlib version check in `pandas.util.testing`, but I can't run code from that in the main body of `pandas.tools.plotting` (I'm assuming a circular import).

Comment (TomAugspurger) [inline pandas/core/config_init.py]: Bah, this doesn't work since you can't set the option multiple times.

File: pandas/tools/plotting.py [hunk 5]
```diff
@@ -24,13 +24,18 @@
...
 from pandas.util.decorators import Appender
-
 try:  # mpl optional
     import pandas.tseries.converter as conv
     conv.register()  # needs to override so set_xlim works with str/number
 except ImportError:
     pass
 
+def _mpl_ge_1_5_0():
+    import matplotlib
```
Comment (jreback) [inline pandas/tools/plotting.py]: seems you already have this in ``utils.testing``?

File: pandas/tseries/tests/test_plotting.py [hunk 6]
```diff
@@ -486,9 +486,9 @@ def test_gaps(self):
         self.assertEqual(len(lines), 1)
         l = lines[0]
         data = l.get_xydata()
-        tm.assertIsInstance(data, np.ma.core.MaskedArray)
-        mask = data.mask
-        self.assertTrue(mask[5:25, 1].all())
+        # tm.assertIsInstance(data, np.ma.core.MaskedArray)
+        # mask = data.mask
```
Comment (jreback) [inline pandas/tseries/tests/test_plotting.py]: these just don't work? put skips rather than comment out then

File: pandas/util/testing.py [hunk 7]
```diff
@@ -41,6 +41,7 @@
 from pandas.util.decorators import deprecate
 from pandas import _testing
 from pandas.io.common import urlopen
+from pandas.tools.plotting import _mpl_ge_1_5_0
```
Comment (jreback) [inline pandas/util/testing.py]: don't do this, just define it here

**Comment by jreback** [comment]
is this not tested on travis? I don't see the icon

File: ci/install_conda.sh [hunk 8]
```diff
@@ -72,6 +72,7 @@ bash miniconda.sh -b -p $HOME/miniconda || exit 1
 
 conda config --set always_yes yes --set changeps1 no || exit 1
 conda update -q conda || exit 1
+conda config --add channels tacaswell || exit 1
```
Comment (tacaswell) [inline ci/install_conda.sh]: the conda-forge channel might be better to use here

**Comment by TomAugspurger** [comment]
Travis was flaky yesterday morning. I just pushed some changes, and it's picked up now.

**Comment by TomAugspurger** [comment]
@jreback travis is green. Looks like https://travis-ci.org/pydata/pandas/jobs/81429896 and https://travis-ci.org/pydata/pandas/jobs/81429897 both ran with matplotlib RC1.

**Comment by jreback** [comment]
make the 3.4 regular build not specify a matplotlib (in requirements-3.4.run). All of the slow tests are running (e.g. 2.7) just fine. But I don't think the fast ones actually are run by travis (in theory they are on 3.3, but that is not build ATM for 1.5.0rc1).

**Comment by TomAugspurger** [comment]
@jreback done. We'll want to followup and pin it at 1.5 once that's released. I'm assuming the `conda-forge` channel will continue to track master and we won't want changes there breaking our builds.

**Comment by jreback** [comment]
thanks @TomAugspurger

**Comment by tacaswell** [comment]
@TomAugspurger I don't think we plan to update conda forge from master, just tags.

**Pull request merged**